### ISSUE TITLE
bNotificationAuthenticationComplete sometimes not being called

### ISSUE THREAD

**Issue by GreenRollingHills** [chat-sdk/chat-sdk-ios#152]
Hello,

So, in my integration of chatSDK I can’t always get my app to post the following Notification: `bNotificationAuthenticationComplete` .  In particular I can’t get the block that it is in to be called.  

It will not post in the following scenario:  When user attempts a firebase anonymous log in from a logged out state.  In other scenarios the notification is posted.

In the chatSDK demo I can get the demo to make this post when user attempts a firebase anonymous log in from a logged out state. 

In my app I call `anonymousButtonPressed` in `ElmLoginViewController` . Then `loginWithFirebaseUser` in `BFirebaseAuthenticationHandler` is eventually called. ` [user once]` is also called in `loginWithFirebaseUser` .  But again, the block that contains the `bNotificationAuthenticationComplete` post isn’t called.

I am not sure what I am doing wrong. Perhaps I can get a clue or 2 as to what might prevent the block from being called? 

Thanks,

**Comment by bensmiley** [comment]
I've tested this and in the default project this notification is always posted. I think this may be a problem with your integration of the library. How are you authenticating the user?

**Comment by GreenRollingHills** [comment]
Hello,

So, I figured it out. sorta.

I am using different images for the user image URLs in my app.  So, when I took out my implementation of setting the initial user imageURL, the  `bNotificationAuthenticationComplete` notification was posted at all appropriate times.  

So, I came up with a different implementation for the initial user imageURL.  I then got the `bNotificationAuthenticationComplete` notification to post at all appropriate times and was properly setting the initial user imageURL.

Not sure exactly what the issue was, but everything is working fine now.
